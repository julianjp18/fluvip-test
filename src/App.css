.image-card{
    height: 200px;
    width: 100%;
    display: block;
}

a:hover{
    text-decoration: none;
}