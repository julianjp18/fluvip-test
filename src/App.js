import React from 'react';
import { BrowserRouter as Router, Switch, Route, Link } from 'react-router-dom';
import './styles/bootstrap.css'; 
import './App.css';
import Pets from './components/Pets';
import Users from './components/Users';
import NewUser from './components/NewUser';
import NewPet from './components/NewPet';
import User from './components/User';
import UserPets from './components/UserPets';
import Pet from './components/Pet';


const homeOptions = () => (
  <div className="container">
    <p className="text-center">Selecciona alguna de las dos opciones:</p> 
    <div className="row">
      <div className="col-md-3"></div>
      <div className="col-md-3">
        <Link to={'/pets'}>
          <div className="card text-white bg-success mb-3">
            <h3 className="card-header text-center">Mascotas</h3>
            <img className="image-card" src="https://images.pexels.com/photos/46024/pexels-photo-46024.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=650&w=940" alt="Pets"></img>
              <div className="card-body">
                <p className="card-text">Visualiza a todas las mascotas de Fluvip Pet</p>
              </div>
          </div>  
        </Link>
      </div>
      <div className="col-md-3">
        <Link to={'/users'}>
          <div className="card text-white bg-dark mb-3">
            <h3 className="card-header text-center">Dueños</h3>
            <img className="image-card" src="https://images.pexels.com/photos/745045/pexels-photo-745045.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=650&w=940" alt="Users"></img>
            <div className="card-body">
              <p className="card-text">Visualiza a todos los dueños de mascotas de Fluvip Pet</p>
            </div>
          </div>
        </Link>
      </div>
      <div className="col-md-3"></div>
    </div>
  </div>
);

function App() {
  return (
    <Router>
          <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
            <Link to={'/'} className="navbar-brand">Fluvip Pet</Link>
            <div className="collapse navbar-collapse" id="navbarSupportedContent">
              <ul className="navbar-nav mr-auto">
              <li className="nav-item">
                  <Link to={'/'} className="nav-link">Inicio</Link>
                </li>
                <li className="nav-item">
                  <Link to={'/users'} className="nav-link">Dueños</Link>
                </li>
                <li className="nav-item">
                  <Link to={'/pets'} className="nav-link">Mascotas</Link>
                </li>
              </ul>
            </div>
          </nav>
          <div className="container">
            <br/>
            <h1 className="text-center">Fluvip Pet</h1>
            <br/>
            <Switch>
            
                <Route exact path='/' component={ homeOptions } />
                <Route exact path='/users' component={ Users } />
                <Route path='/pets' component={ Pets } />
                <Route path='/newUser' component={ NewUser } />
                <Route path='/newPet' component={ NewPet } />
                <Route path='/user' component={ User } />
                <Route path='/userPets' component={ UserPets } />
                <Route path='/pet' component={ Pet } />
            </Switch>
          </div>
      </Router>
  );
}

export default App;
