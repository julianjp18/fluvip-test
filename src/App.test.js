import {React, Counter} from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from './App';

test('renders learn react link', () => {
  const { getByText } = render(<App />);
  const linkElement = getByText(/Visualiza a todas las mascotas de Fluvip Pet/i);
  expect(linkElement).toBeInTheDocument();
});