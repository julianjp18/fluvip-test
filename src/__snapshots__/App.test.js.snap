// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders 1`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Fluvip Pet
    </a>
    <div
      className="collapse navbar-collapse"
      id="navbarSupportedContent"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/"
            onClick={[Function]}
          >
            Inicio
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/users"
            onClick={[Function]}
          >
            Dueños
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/pets"
            onClick={[Function]}
          >
            Mascotas
          </a>
        </li>
      </ul>
    </div>
  </nav>,
  <div
    className="container"
  >
    <br />
    <h1
      className="text-center"
    >
      Fluvip Pet
    </h1>
    <br />
    <div
      className="container"
    >
      <p
        className="text-center"
      >
        Selecciona alguna de las dos opciones:
      </p>
      <div
        className="row"
      >
        <div
          className="col-md-3"
        />
        <div
          className="col-md-3"
        >
          <a
            href="/pets"
            onClick={[Function]}
          >
            <div
              className="card text-white bg-success mb-3"
            >
              <h3
                className="card-header text-center"
              >
                Mascotas
              </h3>
              <img
                alt="Pets"
                className="image-card"
                src="https://images.pexels.com/photos/46024/pexels-photo-46024.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=650&w=940"
              />
              <div
                className="card-body"
              >
                <p
                  className="card-text"
                >
                  Visualiza a todas las mascotas de Fluvip Pet
                </p>
              </div>
            </div>
          </a>
        </div>
        <div
          className="col-md-3"
        >
          <a
            href="/users"
            onClick={[Function]}
          >
            <div
              className="card text-white bg-dark mb-3"
            >
              <h3
                className="card-header text-center"
              >
                Dueños
              </h3>
              <img
                alt="Users"
                className="image-card"
                src="https://images.pexels.com/photos/745045/pexels-photo-745045.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=650&w=940"
              />
              <div
                className="card-body"
              >
                <p
                  className="card-text"
                >
                  Visualiza a todos los dueños de mascotas de Fluvip Pet
                </p>
              </div>
            </div>
          </a>
        </div>
        <div
          className="col-md-3"
        />
      </div>
    </div>
  </div>,
]
`;
